client/pkg/template/internal/models/controlplane.go

```diff
@@ -19,6 +20,12 @@ const KindControlPlane = "ControlPlane"
 // ControlPlane describes Cluster controlplane nodes.
 type ControlPlane struct {
 	MachineSet `yaml:",inline"`
+	Managed    Managed `yaml:"managed,omitempty"`
+}
+
+// Managed describes managed control planes mode.
+type Managed struct {
+	Enable bool `yaml:"enable"`
```

Comment:
I don't quite see how managed controlplane can be in a non-managed cluster ?

Reply:
probably we should name it differently: should indicate that the machines are allocated from our infra.